src/payment/data/service.js

```diff
@@ -32,6 +33,13 @@ export function configureApiService(newConfig, newApiClient) {
   apiClient.interceptors.response.use((response) => {
     const { status, data } = response;
     if (status >= 200 && status < 300 && data && data.redirect) {
+      // Redirecting this SPA to itself is likely to cause
+      // a redirect loop.
+      if (window.location.href === data.redirect) {
```

Comment:
Do we want global instead?

Reply:
I think* it should not be an issue even in a gatsby world since this will never fire until runtime. I'll update anyway.